fecgov/openFEC
# ADR additional parameters

## Summary (required)

- Resolves #4793 

Add ADR additional parameters:
 - Complainant
-  Case status
-  Non-monetary terms/non-civil penalties
-  Commission votes: commissioner names and vote
-  Citations 

### Required reviewers

- 1 back end developer and @JonellaCulmer 

## Impacted areas of the application

-  ADR 


## How to test
- Run: `git checkout feature/4793-adr-parameters`
- Run: `pytest` (All tests PASS)
- Setup local Elasticsearch service and test on local env: Follow [Elasticsearch-7.x.0-setup-and-test-locally]( https://github.com/fecgov/openFEC/wiki/Elasticsearch-7.x.0-setup-and-test-locally)
- Delete existing indexes, run: `python cli.py delete_index docs`
- Create `docs` indexes, run:  `python cli.py create_index`
- Verify elasticsearch mappings: `http://localhost:9200/_mappings` (on browser, adr mappings will show)
- Load adr(s), run:  `python cli.py load_adrs 123` , `python cli.py load_adrs 333`,  `python cli.py load_adrs 900` or your choice
- http://localhost:9200/docs/_doc/adr_123
- http://localhost:9200/docs/_doc/adr_333
- http://localhost:9200/docs/_doc/adr_900
- compare local es data vs EQS data 

** Notes: If this PR is deployed  to `Dev` space then consider 
1. delete exiting indexes
2. create new docs index
3. load legal docs onto `Dev` space

Screenshots of ADR data on `EQS` vs `PG` vs `feature/4793-adr-parameters`:

###  ADR#078:
### **Citations:**

**EQS:**

<img width="705" alt="Screen Shot 2022-10-12 at 8 47 15 PM" src="https://user-images.githubusercontent.com/11650355/195473672-5bcfd8c6-d670-4571-9a65-f9fbe865c1ed.png">

**PG:**
Query:
```
SELECT DISTINCT VIOLATIONS.statutory_citation, VIOLATIONS.regulatory_citation, entity.name, entity.entity_id
    FROM  fecmur.VIOLATIONS, fecmur.ENTITY
    WHERE VIOLATIONS.case_id = 36000000000181
    AND VIOLATIONS.ENTITY_ID = ENTITY.ENTITY_ID
    AND (case when trim(VIOLATIONS.STAGE) = '' or VIOLATIONS.STAGE is null then 'None'
		     else upper(VIOLATIONS.STAGE)
	end ) in (select case when trim(STAGE_ORDER.STAGE) = '' or STAGE_ORDER.STAGE is null then 'None'
		                  else upper(STAGE_ORDER.STAGE)
			         end
			  from fecmur.STAGE_ORDER)
    ORDER BY entity.name

```
<img width="949" alt="Screen Shot 2022-10-12 at 8 50 27 PM" src="https://user-images.githubusercontent.com/11650355/195473918-07a2e46a-375b-4ac8-93d0-d65ebce65b04.png">

**local api -> legal/docs endpoint:**
<img width="579" alt="Screen Shot 2022-10-12 at 8 47 56 PM" src="https://user-images.githubusercontent.com/11650355/195473682-bc62056b-bc5d-4600-8943-ae386677f892.png">

---------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------


### **Commission votes: commissioner names and vote**

**EQS:**
<img width="703" alt="Screen Shot 2022-10-12 at 8 52 57 PM" src="https://user-images.githubusercontent.com/11650355/195474338-13133e3f-7a29-4861-bee9-bba99c76aa79.png">


**PG:**

Query:
    
```
SELECT DISTINCT c."action", c.vote_date , e.name as commissioner_name,  v.vote_type  
    FROM fecmur.COMMISSION c, fecmur.votes v, fecmur.entity e
    WHERE c.CASE_ID = 36000000000181
    AND c.commission_id = v.commission_id
    AND e.entity_id =  v.entity_id
    ORDER BY c.vote_date desc;

```
<img width="1094" alt="Screen Shot 2022-10-12 at 8 55 57 PM" src="https://user-images.githubusercontent.com/11650355/195474512-dc33b27a-6230-4c08-a493-6efac2589342.png">


**local api -> legal/docs endpoint:**

<img width="1426" alt="Screen Shot 2022-10-12 at 8 53 17 PM" src="https://user-images.githubusercontent.com/11650355/195474540-79538c32-10b5-46e8-b1dc-271d98b5bb0a.png">

----------------------------------------------------------------------------------------------------------------------------
----------------------------------------------------------------------------------------------------------------------------

### Non-monetary terms:

**EQS:**
<img width="339" alt="Screen Shot 2022-10-12 at 9 03 55 PM" src="https://user-images.githubusercontent.com/11650355/195475418-037743a3-8f7e-45b4-9243-42ff0222c300.png">

**PG:**
Query:

   ```
SELECT DISTINCT nt.term_description, nt.term_id
    FROM fecmur.case c, fecmur.SETTLEMENT s, fecmur.RELATEDOBJECTS r, fecmur.NON_MONETARY_TERM NT
    WHERE c.case_id  = 36000000000181
    AND c.case_id = s.case_id
    AND UPPER (SETTLEMENT_TYPE) LIKE 'PENALTY'
    AND r.RELATION_ID = 8
    AND r.MASTER_KEY = s.settlement_id
    AND R.DETAIL_KEY = NT.TERM_ID
```

<img width="881" alt="Screen Shot 2022-10-12 at 9 06 47 PM" src="https://user-images.githubusercontent.com/11650355/195475633-67328e3a-ec62-4f6c-8cc1-3549d192d2a7.png">


**local api -> legal/docs endpoint:**

<img width="610" alt="Screen Shot 2022-10-12 at 9 04 27 PM" src="https://user-images.githubusercontent.com/11650355/195475438-ab928ae1-db5a-48fd-89e5-20fa19233755.png">
---------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------

**Case status is populated on API based on disposition description:** 
- There is no case_status field on EQS

**local api -> legal/docs endpoint:**
<img width="536" alt="Screen Shot 2022-10-17 at 10 37 16 AM" src="https://user-images.githubusercontent.com/11650355/196206512-5f0c41b1-8069-4acd-af74-a51231f5119b.png">


----------------------------------------------------------------------------------------------------------------------------
----------------------------------------------------------------------------------------------------------------------------

### Complainant/Referring office: 

**EQS:**

<img width="438" alt="Screen Shot 2022-10-12 at 9 23 17 PM" src="https://user-images.githubusercontent.com/11650355/195477262-6e23b6a2-68ac-4de4-9fd7-7d0a4659b301.png">

**PG:**
Query:
```
SELECT name as complainant_name
   FROM fecmur.players 
   JOIN fecmur.role USING (role_id)
   JOIN fecmur.entity USING (entity_id)
   WHERE  players.case_id = %s
   AND players.role_id IN (100095)
```
<img width="376" alt="Screen Shot 2022-10-12 at 9 24 52 PM" src="https://user-images.githubusercontent.com/11650355/195477436-14c06868-b25e-4ca5-915e-3f8e39288c39.png">


**local api -> legal/docs endpoint:**
<img width="294" alt="Screen Shot 2022-10-12 at 9 19 09 PM" src="https://user-images.githubusercontent.com/11650355/195476996-79b7dfbf-f9e2-49fb-8d4f-098d1cc4c903.png">

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
